lib/determinator/cache/fetch_wrapper.rb

```diff
@@ -1,20 +1,49 @@
 module Determinator
   module Cache
     class FetchWrapper
-      # @param cache [#fetch] An instance of a cache class which implements #fetch like ActiveSupport::Cache does
-      def initialize(cache)
-        @cache = cache
+      # @param *caches [#fetch] n instances of an ActiveSupport::Cache (or eqivelent)
+      def initialize(*caches)
+        @caches = caches
       end
 
+      # Call walks through each cache, returning a value if the item exists in
+      # any cache, otherwise popularing each cache with the value of yield.
+
       def call(feature_name)
-        @cache.fetch(key(feature_name)) { yield }
+        value = read_and_upfill(feature_name)
+        # Nil is an acceptable value, do not yield
+        return nil if value.nil?
+        return value if value != false
+
+        value_to_write = yield
+        @caches.each do |cache|
+          cache.write(key(feature_name), value_to_write)
+        end
+        return value_to_write
+      end
+
+      def expire(feature_name)
+        @caches.each{ |c| c.delete(key(feature_name)) }
       end
 
       private
 
       def key(feature_name)
         "determinator:feature_cache:#{feature_name}"
       end
+
+      def read_and_upfill(feature_name)
```

Comment:
Given that `false` and `nil` have specific meaning here, might be worth documenting them in this method's YARDoc?